TimeoutException/DnsNameResolverTimeoutException on sending event to a sink

**Describe the bug**
```
{"@timestamp":"2021-01-12T13:33:22.128Z","@version":"1","message":"Failed to send event to subscriber topic=knative-broker-test-broker-consumer-v1-beta1-1-sx6vh-broker partition=5 headers=[io.vertx.kafka.client.producer.impl.KafkaHeaderImpl@17aac7cb, io.vertx.kafka.client.producer.impl.KafkaHeaderImpl@16b20795, io.vertx.kafka.client.producer.impl.KafkaHeaderImpl@5f1f85dc, io.vertx.kafka.client.producer.impl.KafkaHeaderImpl@443791e7, io.vertx.kafka.client.producer.impl.KafkaHeaderImpl@5629235] offset=0 event=CloudEvent{id='no-upgrade', source=consumer-test-sender, type='dev.knative.test.event', datacontenttype='application/json', schemaurl=null, subject='null', time=null, data=BytesCloudEventData{value=[123, 34, 109, 115, 103, 34, 58, 34, 99, 111, 110, 115, 117, 109, 101, 114, 45, 98, 114, 111, 107, 101, 114, 45, 116, 101, 115, 116, 115, 34, 125]}, extensions{}}","logger_name":"dev.knative.eventing.kafka.broker.dispatcher.ConsumerRecordHandler","thread_name":"vert.x-eventloop-thread-0","level":"ERROR","level_value":40000,"stack_trace":"io.vertx.circuitbreaker.TimeoutException: operation timeout\n","topic":"knative-broker-test-broker-consumer-v1-beta1-1-sx6vh-broker","partition":5,"headers":[{},{},{},{},{}],"offset":0,"event":{"data":{},"id":"no-upgrade","source":"consumer-test-sender","type":"dev.knative.test.event","subject":null,"time":null,"schemaUrl":null,"dataContentType":"application/json","dataSchema":null,"specVersion":"V03","extensionNames":[],"attributeNames":["datacontenttype","specversion","id","source","type"]}}
```
Sink logs:
```
{"level":"info","ts":"2021-01-12T13:32:40.845Z","caller":"logging/config.go:115","msg":"Successfully created the logger.","logging.googleapis.com/labels":{},"logging.googleapis.com/sourceLocation":{"file":"knative.dev/pkg@v0.0.0-20210107022335-51c72e24c179/logging/config.go","line":"115","function":"knative.dev/pkg/logging.newLoggerFromConfig"}}
{"level":"info","ts":"2021-01-12T13:32:40.845Z","caller":"logging/config.go:116","msg":"Logging level set to: debug","logging.googleapis.com/labels":{},"logging.googleapis.com/sourceLocation":{"file":"knative.dev/pkg@v0.0.0-20210107022335-51c72e24c179/logging/config.go","line":"116","function":"knative.dev/pkg/logging.newLoggerFromConfig"}}
{"level":"info","ts":"2021-01-12T13:32:40.845Z","caller":"logging/config.go:78","msg":"Fetch GitHub commit ID from kodata failed","error":"open /var/run/ko/HEAD: no such file or directory","logging.googleapis.com/labels":{},"logging.googleapis.com/sourceLocation":{"file":"knative.dev/pkg@v0.0.0-20210107022335-51c72e24c179/logging/config.go","line":"78","function":"knative.dev/pkg/logging.enrichLoggerWithCommitID"}}
{"level":"info","ts":"2021-01-12T13:32:40.846Z","logger":"recordevents","caller":"recorder_vent/constructor.go:53","msg":"Recorder vent environment configuration: {AgentName:observer-default PodName:transformer-pod Port:8080}","logging.googleapis.com/labels":{},"logging.googleapis.com/sourceLocation":{"file":"knative.dev/eventing@v0.19.1-0.20210106210736-1ed0e63c3887/test/lib/recordevents/recorder_vent/constructor.go","line":"53","function":"knative.dev/eventing/test/lib/recordevents/recorder_vent.NewFromEnv"}}
{"level":"info","ts":"2021-01-12T13:32:40.885Z","logger":"recordevents","caller":"recorder_vent/constructor.go:64","msg":"Going to send events to pod 'transformer-pod' in namespace 'test-broker-consumer-v1-beta1-1-sx6vh'","logging.googleapis.com/labels":{},"logging.googleapis.com/sourceLocation":{"file":"knative.dev/eventing@v0.19.1-0.20210106210736-1ed0e63c3887/test/lib/recordevents/recorder_vent/constructor.go","line":"64","function":"knative.dev/eventing/test/lib/recordevents/recorder_vent.NewEventLog"}}
{"level":"info","ts":"2021-01-12T13:32:40.886Z","logger":"recordevents","caller":"receiver/receiver.go:86","msg":"Receiver environment configuration: {ReceiverName:transformer-pod Reply:true ReplyEventType:reply-check-type ReplyEventSource:reply-check-source ReplyEventData:{\"msg\":\"Transformed!\"} ReplyAppendData: SkipStrategy: SkipCounter:0}","logging.googleapis.com/labels":{},"logging.googleapis.com/sourceLocation":{"file":"knative.dev/eventing@v0.19.1-0.20210106210736-1ed0e63c3887/test/lib/recordevents/receiver/receiver.go","line":"86","function":"knative.dev/eventing/test/lib/recordevents/receiver.NewFromEnv"}}
{"level":"info","ts":"2021-01-12T13:32:40.886Z","logger":"recordevents","caller":"receiver/receiver.go:90","msg":"Receiver will reply with an event","logging.googleapis.com/labels":{},"logging.googleapis.com/sourceLocation":{"file":"knative.dev/eventing@v0.19.1-0.20210106210736-1ed0e63c3887/test/lib/recordevents/receiver/receiver.go","line":"90","function":"knative.dev/eventing/test/lib/recordevents/receiver.NewFromEnv"}}
```


 **Expected behavior**

No timeout exception.

 **To Reproduce**

Run E2E tests in CI.

 **Knative release version**
`<= devel`

 **Additional context**

- artifacts of https://prow.knative.dev/view/gcs/knative-prow/logs/ci-knative-sandbox-eventing-kafka-broker-continuous/1348977869178540032

It seems related to a DNS issue:
```
{"@timestamp":"2021-01-12T13:33:22.129Z","@version":"1","message":"Failed to send event to subscriber topic=knative-broker-test-event-transformation-for-trigger-v1-broker-v1-beta1-48p7xn-broker partition=4 headers=[io.vertx.kafka.clie    nt.producer.impl.KafkaHeaderImpl@539931c5, io.vertx.kafka.client.producer.impl.KafkaHeaderImpl@45532fad, io.vertx.kafka.client.producer.impl.KafkaHeaderImpl@42bfbd62, io.vertx.kafka.client.producer.impl.KafkaHeaderImpl@4bb7296c, io.ve    rtx.kafka.client.producer.impl.KafkaHeaderImpl@51ba2ec6] offset=0 event=CloudEvent{id='b264196e-2608-46b9-aed7-717e6235d067', source=source1, type='type1', datacontenttype='application/json', dataschema=null, subject='null', time=null    , data=BytesCloudEventData{value=[123, 34, 109, 115, 103, 34, 58, 34, 101, 50, 101, 45, 101, 118, 101, 110, 116, 116, 114, 97, 110, 115, 102, 111, 114, 109, 97, 116, 105, 111, 110, 45, 98, 111, 100, 121, 34, 125]}, extensions={}}","lo    gger_name":"dev.knative.eventing.kafka.broker.dispatcher.ConsumerRecordHandler","thread_name":"vert.x-eventloop-thread-0","level":"ERROR","level_value":40000,"stack_trace":"java.net.UnknownHostException: failed to resolve 'trans-pod.t    est-event-transformation-for-trigger-v1-broker-v1-beta1-48p7xn.svc.cluster.local'\n\tat io.netty.resolver.dns.DnsResolveContext.finishResolve(DnsResolveContext.java:927)\n\tat io.netty.resolver.dns.DnsResolveContext.tryToFinishResolve    (DnsResolveContext.java:886)\n\tat io.netty.resolver.dns.DnsResolveContext.query(DnsResolveContext.java:358)\n\tat io.netty.resolver.dns.DnsResolveContext.access$600(DnsResolveContext.java:64)\n\tat io.netty.resolver.dns.DnsResolveCon    text$2.operationComplete(DnsResolveContext.java:407)\n\tat io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577)\n\tat io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:570)\n\tat     io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:549)\n\tat io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490)\n\tat io.netty.util.concurrent.DefaultPromise.setValue0(Default    Promise.java:615)\n\tat io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:608)\n\tat io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117)\n\tat io.netty.resolver.dns.DnsQueryContext.tryFail    ure(DnsQueryContext.java:225)\n\tat io.netty.resolver.dns.DnsQueryContext$4.run(DnsQueryContext.java:177)\n\tat io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)\n\tat io.netty.util.concurrent.ScheduledFutureTask.run(S    cheduledFutureTask.java:170)\n\tat io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)\n\tat io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)\n\t    at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:497)\n\tat io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)\n\tat io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.    java:74)\n\tat io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)\n\tat java.base/java.lang.Thread.run(Unknown Source)\nCaused by: io.netty.resolver.dns.DnsNameResolverTimeoutException: [/10.75.240.1    0:53] query via UDP timed out after 5000 milliseconds (no stack trace available)\n","topic":"knative-broker-test-event-transformation-for-trigger-v1-broker-v1-beta1-48p7xn-broker","partition":4,"headers":[{},{},{},{},{}],"offset":0,"e    vent":{"data":{},"id":"b264196e-2608-46b9-aed7-717e6235d067","source":"source1","type":"type1","subject":null,"time":null,"dataContentType":"application/json","dataSchema":null,"specVersion":"V1","extensionNames":[],"attributeNames":[    "datacontenttype","specversion","id","source","type"]}}
```

--- a/data-plane/config/template/500-dispatcher.yaml
+++ b/data-plane/config/template/500-dispatcher.yaml
@@ -139,3 +139,6 @@ spec:
           configMap:
             name: config-tracing
       restartPolicy: Always
+      dnsConfig:
+        options:
+          - name: single-request-reopen

